### PR TITLE
Make nsga2 tests optional + docs update

### LINKED ISSUE #345: NSGA2 tests should be optional
## Description

NSGA2 is now an optional optimizer, since #300. Even though we provide the source, we only install it if SWIG is available. As a result, NSGA2-related tests should be skipped if it is not available.

### PR DESCRIPTION
**Pull request by nwu63**
## Purpose

Closes #345. Also updated docs slightly + added an nsga2 test to the Rosenbrock problem.

## Expected time until merged

1 week

## Type of change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (non-backwards-compatible fix or feature)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [x] Documentation update
- [x] Maintenance update
- [ ] Other (please describe)

## Checklist

- [x] I have run `flake8` and `black` to make sure the Python code adheres to PEP-8 and is consistently formatted
- [ ] I have formatted the Fortran code with `fprettify` or C/C++ code with `clang-format` as applicable
- [x] I have run unit and regression tests which pass locally with my changes
- [ ] I have added new tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation

### CODE REVIEW

**Review by sseraj — APPROVED** [review]

**Comment by nwu63** [comment]
Bump @lamkina

File: tests/test_nsga2_multi_objective.py [hunk 1]
```diff
@@ -43,6 +50,9 @@ def setup_optProb(self, n_obj):
         if n_obj == 2:
             self.optProb.addObj("obj2")
 
+    @unittest.skipIf(
```
Comment (marcomangano) [inline tests/test_nsga2_multi_objective.py:53]: I am sorry to have checked this too late @nwu63 , but I don't really get the point of this step here. Can't we remove this altogether?  We are already skipping all `n_obj == 2` tests by default on all platforms. Is this related to an additional test that `testflo` runs on windows?
Locally I only have the last two tests of this stack:
```
about to run test_nsga2_multi_objective.py:TestNSGA2.test_opt .\test_nsga2_multi_objective.py:TestNSGA2.test_opt  ... SKIP (00:00:0.00, 0 MB)
test_nsga2_multi_objective.py fails on windows with two objectives! Skipping for now.
    about to run test_nsga2_multi_objective.py:TestNSGA2.test_opt_0 .\test_nsga2_multi_objective.py:TestNSGA2.test_opt_0 ... OK (00:00:4.17, 0 MB)
    about to run test_nsga2_multi_objective.py:TestNSGA2.test_opt_1 .\test_nsga2_multi_objective.py:TestNSGA2.test_opt_1  ... SKIP (00:00:0.00, 0 MB)
skip flaky NSGA2 tests
```

Comment (nwu63) [reply tests/test_nsga2_multi_objective.py:53]: This skips all tests on Windows, the below skips all multiobjective tests, so the only tests being run are the single-objective tests on linux.

Comment (marcomangano) [reply tests/test_nsga2_multi_objective.py:53]: I understand the idea (and that is what I was expecting) but that printout in my comment is from the Windows test log - apparently the single-objective test is running on Windows? Or is it a Testflo fluke?

Comment (nwu63) [reply tests/test_nsga2_multi_objective.py:53]: It is possible that the skipIf doesn't play nicely with parameterized. _However_ this is actually what we want---the single objective test should pass on Windows and we should test it, while the multi-objective test appears to be flat out broken. Let me throw together a PR to update this.

**Pull request merged**